src/ja/server/dispatcher/dispatcher.py

```diff
@@ -36,17 +38,35 @@ def set_distribution(self, job_distribution: JobDistribution) -> None:
                 proxy = self._proxy_factory.get_proxy(work_machine)
                 if job.status == JobStatus.RUNNING:
                     if previous_status is None or previous_status == JobStatus.QUEUED:
-                        proxy.dispatch_job(job)
+                        try:
+                            proxy.dispatch_job(job)
+                        except Exception:
```

Comment:
This catch is too general.
Please replace it with the actual exception time thrown upon timeout.